File: python/2006.差的绝对值为k的数对数目.py
class Solution:
    # [题解]1. O(n) t:48ms(76%) O(n) m:14.9M(72%) 哈希
    def countKDifference(self, nums: List[int], k: int) -> int:
        cnts = Counter()
        ans = 0
        for i in nums:
            ans += cnts[i - k] + cnts[i + k]
            cnts[i] += 1
        return ans

File: java/2006.差的绝对值为k的数对数目.java
class Solution {
    // [题解]1. O(n) t:4ms(75%) O(n) m:40.4M(11%)
    public int countKDifference(int[] nums, int k) {
        Map<Integer, Integer> cnts = new HashMap<Integer, Integer>();
        int ans = 0;
        for (int i: nums){
            ans += cnts.getOrDefault(i - k, 0) + cnts.getOrDefault(i + k, 0);
            cnts.put(i, cnts.getOrDefault(i, 0) + 1);
        }
        return ans;
    }
}